robot/impl/local_robot_test.go

```diff
@@ -3907,3 +3908,310 @@ func TestLogPropagation(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckMaintenanceSensorReadings(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+	t.Run("Sensor reading errors out", func(t *testing.T) {
+		r := setupLocalRobot(t, context.Background(), &config.Config{}, logger)
+		localRobot := r.(*localRobot)
+		canReconfigure, err := localRobot.checkMaintenanceSensorReadings(context.Background(), "", newErrorSensor())
+
+		test.That(t, canReconfigure, test.ShouldEqual, false)
+		test.That(t, err.Error(), test.ShouldEqual, "error reading maintenance sensor readings. Wallet not found")
+	})
+	t.Run("maintenanceAllowedKey does not exist", func(t *testing.T) {
+		r := setupLocalRobot(t, context.Background(), &config.Config{}, logger)
+		localRobot := r.(*localRobot)
+		canReconfigure, err := localRobot.checkMaintenanceSensorReadings(context.Background(), "keyDoesNotExist", newValidSensor())
+
+		test.That(t, canReconfigure, test.ShouldEqual, true)
+		test.That(t, err.Error(), test.ShouldEqual, "error getting maintenance_allowed_key keyDoesNotExist from sensor reading")
+	})
+	t.Run("maintenanceAllowedKey is not a boolean", func(t *testing.T) {
+		r := setupLocalRobot(t, context.Background(), &config.Config{}, logger)
+		localRobot := r.(*localRobot)
+		canReconfigure, err := localRobot.checkMaintenanceSensorReadings(context.Background(), "ThatIsNotAWallet", newValidSensor())
+
+		test.That(t, canReconfigure, test.ShouldEqual, true)
+		test.That(t, err.Error(), test.ShouldEqual, "maintenanceAllowedKey ThatIsNotAWallet is not a bool value")
+	})
+}
+
+func TestCheckMaintenanceSensorReadingsSuccess(t *testing.T) {
+	logger := logging.NewTestLogger(t)
+	testsValid := []struct {
+		testName              string
+		canReconfigure        bool
+		maintenanceAllowedKey string
+		sensor                resource.Sensor
+	}{
+		{
+			testName:              "Sensor returns reading false",
+			canReconfigure:        false,
+			maintenanceAllowedKey: "ThatsMyWallet",
+			sensor:                newValidSensor(),
+		},
+		{
+			testName:              "Sensor returns reading true",
+			canReconfigure:        true,
+			maintenanceAllowedKey: "ThatsNotMyWallet",
+			sensor:                newValidSensor(),
+		},
+	}
+	for _, tc := range testsValid {
+		t.Run("", func(t *testing.T) {
+			r := setupLocalRobot(t, context.Background(), &config.Config{}, logger)
+			localRobot := r.(*localRobot)
+			canReconfigure, err := localRobot.checkMaintenanceSensorReadings(context.Background(), tc.maintenanceAllowedKey, tc.sensor)
+
+			test.That(t, canReconfigure, test.ShouldEqual, tc.canReconfigure)
+			test.That(t, err, test.ShouldBeNil)
+		})
+	}
+}
+
+func newValidSensor() sensor.Sensor {
+	s := &inject.Sensor{}
+	s.ReadingsFunc = func(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error) {
+		return map[string]any{"ThatsMyWallet": false, "ThatsNotMyWallet": true, "ThatIsNotAWallet": 5}, nil
```

Comment:
since most applications of this will be done through a sensor client, can we stick this map through `protoutils.ReadingGoToProto(readings)` and `protoutils.ReadingProtoToGo(readings)` so we have more confidence this works E2E.

can we add a case for "true" and 1 as well

Reply:
added the test cases